### ISSUE TITLE
Event processing bug?

### ISSUE THREAD

**Issue by sf-issues** [ioam/topographica#238]
Converted from SourceForge issue 3311927, submitted by ceball
Submit Date: 2011-06-05 14:40 GMT

This network:

topo.sim['Activity'] = sheet.CFSheet()

topo.sim.connect(
   'Activity','Activity',delay=1.0,name='Lateral',
   connection_type=projection.CFProjection)

Start event processing:

topo_t000000.00_c1>>> topo.sim["Activity"].send_output(src_port="Activity",data=topo.sim["Activity"].activity)
topo_t000000.00_c2>>>

Running topographica with -d, you can see the events. First one happens:

topo_t000000.00_c2>>> topo.sim.run(1)
Time: 000001.00 r1.py: Delivering EPConnectionEvent(time=mpq(1),conn=CFProjection(activity_group=(0.5, <ufunc 'add'>), allow_null_cfs=False, apply_output_fns_init=True, autosize_mask=True, cf_shape=Constant(bounds=BoundingBox(radius=0.5), mask=None, mask_shape=None, name='Constant00035', offset=0.0, orientation=0.0, output_fns=[], position=[0.0, 0.0], print_level=100, scale=1.0, size=0.30000000000000004, x=0.0, xdensity=10, y=0.0, ydensity=10), cf_type=<class 'topo\.base\.cf\.ConnectionField'>, coord_mapper=IdentityMF(name='IdentityMF00038', print_level=100), delay=1.0, dest=CFSheet(allow_skip_non_responding_units=True, apply_output_fns=True, layout_location=(-1, -1), mask=SheetMask(name='SheetMask00032', print_level=100), measure_maps=True, name='Activity', nominal_bounds=BoundingBox(radius=0.5), nominal_density=10, output_fns=[], plastic=True, precedence=0.5, print_level=100, row_precedence=0.5), dest_port='Activity', learning_fn=CFPLF_Plugin(name='CFPLF_Plugin00037', print_level=100, single_cf_fn=Hebbian(name='Hebbian00017', print_level=100)), learning_rate=0.0, mask_threshold=0.5, min_matrix_radius=1, name='Lateral', nominal_bounds_template=BoundingBox(radius=0.1), output_fns=[], plastic=True, precedence=0.80000000000000004, print_level=100, private=False, response_fn=CFPRF_Plugin(name='CFPRF_Plugin00036', print_level=100, single_cf_fn=DotProduct(name='DotProduct00013', print_level=100)), same_cf_shape_for_all_cfs=True, src=CFSheet(allow_skip_non_responding_units=True, apply_output_fns=True, layout_location=(-1, -1), mask=SheetMask(name='SheetMask00032', print_level=100), measure_maps=True, name='Activity', nominal_bounds=BoundingBox(radius=0.5), nominal_density=10, output_fns=[], plastic=True, precedence=0.5, print_level=100, row_precedence=0.5), src_port='Activity', strength=1.0, weights_generator=Constant(bounds=BoundingBox(radius=0.5), mask=None, mask_shape=None, name='Constant00034', offset=0.0, orientation=0.0, output_fns=[], position=[0.0, 0.0], print_level=100, scale=1.0, size=1.0, x=0.0, xdensity=10, y=0.0, ydensity=10), weights_output_fns=[CFPOF_Plugin(name='CFPOF_Plugin00018', print_level=100, single_cf_fn=IdentityTF(name='IdentityTF00019', norm_value=None, print_level=100))]))
Time: 000001.00 Activity: Received input from Activity on dest_port Activity via connection Lateral.

But there are no subsequent events:

topo_t000001.00_c3>>> topo.sim.run(1)
topo_t000002.00_c4>>> 

However, putting a single future event into the queue results in EPConnectionEvents happening every iteration as expected (at least until the added event is removed from the queue):

topo_t000000.00_c1>>> from topo.base.simulation import FunctionEvent
topo_t000000.00_c2>>> topo.sim.enqueue_event(FunctionEvent(10,lambda:1))
topo_t000000.00_c3>>> topo.sim["Activity"].send_output(src_port="Activity",data=topo.sim["Activity"].activity)
topo_t000000.00_c4>>> topo.sim.run(1)
Time: 000001.00 r1.py: Delivering EPConnectionEvent(time=mpq(1),...
Time: 000001.00 Activity: Received input from Activity on dest_port Activity via connection Lateral.
topo_t000001.00_c5>>> topo.sim.run(1)
Time: 000002.00 r1.py: Delivering EPConnectionEvent(time=mpq(2),...
Time: 000002.00 Activity: Received input from Activity on dest_port Activity via connection Lateral.
...
topo_t000009.00_c7>>> topo.sim.run(1)
Time: 000010.00 r1.py: Delivering FunctionEvent(10,<function <lambda> at 0x8a5448c>,*(),**{})
Time: 000010.00 r1.py: Delivering EPConnectionEvent(time=mpq(10),...
Time: 000010.00 Activity: Received input from Activity on dest_port Activity via connection Lateral.
topo_t000010.00_c8>>> topo.sim.run(1)
topo_t000011.00_c9>>>
